### PR TITLE
Prow needs a timeout.

### LINKED ISSUE #5400: prow stops syncing
Seems like all the PRs cannot be triggered to build now...

/cc @cjwagner @krzyzacy

### PR DESCRIPTION
**Pull request by cjwagner**
Adds a 30 second timeout to all api-server requests. This timeout applies to each of the individual retries of a request not the request as a whole.
Hopefully fixes #5400 
/cc @ixdy 

I picked 30 seconds pretty arbitrarily. Let me know if I should change it.

### CODE REVIEW

**Comment by BenTheElder** [comment]
We should definitely revisit the `bump.sh` process in some other PR :-)

In the meantime, we should roll this out ASAP and see if this fixes things up, cc @krzyzacy

**Comment by krzyzacy** [comment]
`bazel run EVERYTHING`
each sync takes ~10s now, I think 30s is fine? We can try it out.

**Comment by BenTheElder** [comment]
/area prow
/hold cancel
this should be fine, let's get it deployed and see how it goes.

File: prow/kube/client.go [hunk 1]
```diff
@@ -37,7 +37,9 @@ const (
 	inClusterBaseURL = "https://kubernetes.default"
 	maxRetries       = 8
 	retryDelay       = 2 * time.Second
-	EmptySelector    = ""
+	requestTimeout   = time.Second * 30
+
+	EmptySelector = ""
```
Comment (spxtr) [inline prow/kube/client.go]: EmptySelector?

**Comment by spxtr** [comment]
/hold
30 seconds might be a bit short since we regularly hit 10 seconds listing all prow jobs.

**Comment by BenTheElder** [comment]
@cjwagner bump the timeout then let's roll this out?

File: prow/kube/client.go (hunk 1 above)
Comment (kargakis) [reply prow/kube/client.go]: In [apimachinery](https://github.com/kubernetes/apimachinery/blob/18a564baac720819100827c16fdebcadb05b2d0d/pkg/labels/selector.go#L68) this is called a nothingSelector. I should have probably picked the same name. We have copied [k8s.io/apimachinery/pkg/labels](https://github.com/kubernetes/apimachinery/tree/18a564baac720819100827c16fdebcadb05b2d0d/pkg/labels) in [k8s.io/test-infra/prow/kube/labels](https://github.com/kubernetes/test-infra/tree/master/prow/kube/labels).

**Comment by cjwagner** [comment]
Bumped the timeout to 1 minute. I'll start updating the components.

**Pull request merged**